### DISCUSSION TITLE
"Use instead" or "superseded by" for deprecated features

### DISCUSSION THREAD

**Discussion by harry-wg** [mdn/mdn-community#546] [Platform] [answered]
When a feature is marked as deprecated, it would greatly help if mdn could include recommendations of better features that replace it.

Part of the problem is that deprecated features will nevertheless come up as the first results in search engines. So, simply marking a feature as deprecated (or obsolete) is no help to man or beast if there is no link to a suggested replacement.

**Comment by teoli2003** [comment]
Yes! I try to do this. But it is not always easy to find, and sometimes there is no replacement (e.g. Document.execCommand for non-clipboard commands)!

I would answer that we will accept any PR, or deal with any issue, that solves this concretely: that is any issue that suggest the missing information (the replacement) will be dealt with.

**Reply by harry-wg** [reply]
> sometimes there is no replacement

Then it would be useful to say that there is currently nothing else that is similar, to avoid everybody wasting their time fruitlessly searching for one.

But I would argue that until there is a replacement, it's maybe premature to mark the feature as deprecated -- unless the feature is actually unsafe to use in the meantime.

**Reply by hamishwillee** [reply]
> Then it would be useful to say that there is currently nothing else that is similar, to avoid everybody wasting their time fruitlessly searching for one.

I agree with this.

I think a big part of the problem is that we have four formal ways to mark deprecation (inline and header macros, bcd, tags) but no standard way to mark "superseded by". If there was a formal way/hint to indicate a replacement I think that authors might at least think about this.

> it's maybe premature to mark the feature as deprecated -- unless the feature is actually unsafe to use in the meantime.

Not so sure on this. There are many cases where features are deprecated with no replacement. Probably the problem is that no one uses deprecated consistently. MDN often means "discouraged", which could be for lots of reasons, including that it doesn't have broad support.

**Comment by bsmth** [comment, marked as answer]
I think work on this is underway, see https://github.com/mdn/rari/pull/311.

Maybe @LeoMcA can confirm. If this covers the use case, I would say we close this discussion as outdated and track progress here:

- [ ] https://github.com/mdn/fred/issues/654

**Reply by LeoMcA** [reply]
Yes work is underway on that, and it should solve for this problem!
